tests/unit/test_base_client.py

```diff
@@ -232,8 +232,10 @@ def test_table_data_client_not_initialized_w_client_info(self):
         self.assertIs(client._table_data_client, table_data_client)
 
     def test_table_data_client_not_initialized_w_client_options(self):
+        from google.api_core.client_options import ClientOptions
+
         credentials = _make_credentials()
-        client_options = mock.Mock()
+        client_options = ClientOptions("endpoint")
```

Comment:
Please pass optional arguments by name, e.g.:
```suggestion
        client_options = ClientOptions(api_endpoint="endpoint")
```